addons/analytic/models/analytic_mixin.py

```diff
@@ -112,3 +112,7 @@ def _apply_analytic_distribution_domain(self, domain):
             ('analytic_distribution_search', leaf[1], leaf[2]) if len(leaf) == 3 and leaf[0] == 'analytic_distribution' and isinstance(leaf[2], str) else leaf
             for leaf in domain
         ]
+
+    def _get_analytic_account_ids(self):
```

Comment:
Suggestion not asking for change, it could be interesting but it's clearly not a must have :) 
```suggestion
    def _get_analytic_account_ids(self) -> List[int]:
```

Reply:
@assellalou @FlorianGilbert I would change it to:
`def _get_analytic_account_ids(self) -> List[int] | TypeError:`

Calling this util method with a record having no analytic_distribution gives you a TypeError (for ... in False) :D

Reply:
Shit :sweat_smile: 
Do you have write a fix for that? :thinking:

Reply:
@nfl-odoo

Reply:
@FlorianGilbert Noop, I just happen to get the TB on my custom code for a client when using this method, I skipped lines with False distributions as a workaround.
If you rework this method, might be worth to return a set instead of a list (?) Or even a recordset of the said accounts (?) instead of a list that might have duplicates.